src/main/java/org/terasology/fluid/system/FluidCommonSystem.java

```diff
@@ -36,14 +47,24 @@
     @In
     private AssetManager assetManager;
 
+    @In
+    private BlockManager blockManager;
+
     /**
      * Initializes fluid resources and textures at launch time.
      */
     @Override
     public void preBegin() {
-        ResourceUrn waterTextureUri = TextureUtil.getTextureUriForColor(Color.BLUE);
-        Texture texture = assetManager.getAsset(waterTextureUri, Texture.class).get();
-
-        fluidRegistry.registerFluid("Fluid:Water", new TextureFluidRenderer(texture, "water"));
+        for (ResourceUrn blockUrn : Assets.list(BlockFamilyDefinition.class)) {
+            Optional<BlockFamilyDefinition> maybeDefinition = Assets.get(blockUrn, BlockFamilyDefinition.class);
+            if (maybeDefinition.isPresent()) {
+                SectionDefinitionData blockData = maybeDefinition.get().getData().getBaseSection();
```

Comment:
this code can be more idiomitic in the usage of `Optional`

```suggestion
            maybeDefinition.ifPresent(definition -> {
                SectionDefinitionData blockData = definition.getData().getBaseSection();
```

See [12 recipes for using the Optional class as it’s meant to be used](https://blogs.oracle.com/javamagazine/12-recipes-for-using-the-optional-class-as-its-meant-to-be-used) and [The Java Optional class: 11 more recipes for preventing null pointer exceptions](https://blogs.oracle.com/javamagazine/the-java-optional-class-11-more-recipes-for-preventing-null-pointer-exceptions) for more information on this topic 😉